src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -63,11 +64,14 @@ public class AzureBlobGrainStorage : IGrainStorage, ILifecycleParticipant<ISiloL
             {
                 var blob = container.GetBlockBlobReference(blobName);
 
-                string json;
-
+                byte[] contents;
                 try
                 {
-                    json = await blob.DownloadTextAsync().ConfigureAwait(false);
+                    using (var stream = new MemoryStream())
+                    {
+                        await blob.DownloadToStreamAsync(stream).ConfigureAwait(false);
+                        contents = stream.ToArray();
```

Comment:
This is not efficient, but it's the same as what's being done under the hood today, same goes for the JSON serialization